### PR TITLE
Split JS global Scene into Scene and Render

### PR DESCRIPTION
**Pull request by zzmp**
- Split render interface into `render-utils/RenderScriptingInterface`
- Grouped most `render::RenderContext` options into `ItemsMeta` and `Tone` subclasses
- Added accessors for `RenderContext` properties

`ItemsMeta` could be named nicer - `ItemsConfig`?

### CODE REVIEW

File: libraries/render/src/render/Engine.h [hunk 1]
```diff
@@ -32,43 +32,80 @@ const int showNetworkStatusFlag = 2;
...
+
+        struct State : public Counter {
+            bool _render = true;
+            bool _cull = true;
+            bool _sort = true;
+
+            Counter _counter{};
+        };
+
+        // TODO: Store state/counter in a map instead of manually enumerated members
```
Comment (zzmp) [inline libraries/render/src/render/Engine.h]: I think storing this in a map or unordered_map will reduce performance, and i think setCounts is called once per render cycle, so I don't know if this is a good idea. @samcake?

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea9bf00623478c765d07ec5309e4683500c151bc available at: https://gist.github.com/56a5f8560ad50c10926f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6691.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6691.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 326d59a6994fab83feb5693f824ef93cd8d80e31 available at: https://gist.github.com/56a5f8560ad50c10926f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6691.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6691.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cdcb2d565d0dc6e6966ba505d0f16db4733c0d3d available at: https://gist.github.com/56a5f8560ad50c10926f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6691.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6691.dmg

File: libraries/render/src/render/Engine.h [hunk 2]
```diff
@@ -32,43 +32,80 @@ const int showNetworkStatusFlag = 2;
...
-    bool _sortOpaque = true;
-    bool _renderOpaque = true;
-    bool _cullTransparent = true;
-    bool _sortTransparent = true;
-    bool _renderTransparent = true;
-
-    int _numFeedOpaqueItems = 0;
-    int _numDrawnOpaqueItems = 0;
-    int _maxDrawnOpaqueItems = -1;
+    struct ItemsMeta {
```
Comment (samcake) [inline libraries/render/src/render/Engine.h]: WE want class only not struct even if it s all public
you can have the member variables without '_' if they are are public

**Comment by hifi-gustavo** [comment]
Build console output for commit: 674bfa4f370780cb892aa5b85c71d8dd8385b7f3 available at: https://gist.github.com/56a5f8560ad50c10926f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6691.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6691.dmg

**Comment by samcake** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 674bfa4f370780cb892aa5b85c71d8dd8385b7f3 available at: https://gist.github.com/56a5f8560ad50c10926f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6691.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6691.dmg

**Pull request merged**